ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
I am confused too ...
I guess the way we number bits differ

```c
    unitptr_t tmp = (uintptr_t)ORTE_LOCAL_JOBID(name.jobid);
     sentinel |= ((tmp << 1) & 0xfffe);
```
where does the local jobid end into ?
- bits 1:15 as I documented ?
- bits 48:62 ?
and btw, is this endian dependent ?